### ISSUE TITLE
Feature Request: Redirect URL Feature

### ISSUE THREAD

**Issue by trymeouteh** [gab-ai-inc/gab-dissenter-extension#86]
I always use invidio.us as a YouTube front end instead of youtube.com. But to always add a comment or view comments to a YouTube video while on invidio.us I need to go to youtube.com to add my comments or view the comments. A cool feature would be to have an option in the settings to choose what sites will redirect and scan another URL for comments.

For example if I am on...

https://invidio.us/watch?v=7vQmywp8Tus

The dissenter extension will get the comments from this URL...

https://www.youtube.com/watch?v=7vQmywp8Tus

This should be optional of course so the user can choose what websites it will do this on and not be enabled by default.

**Comment by justagoodperson** [comment]
Seems like a bit of an edge case, can you think of many more instances where you would want the discussion page to be pointed to a different url?

**Comment by trymeouteh** [comment]
True, another suggestion instead or redirecting would be to have the option for users block certain URLs to prevent them from anciently commenting on them. That way if your watching a video on invidious you don't accidentally comment on there when you meant to leave a comment on the YouTube URL.

The reason why I like invidious is because it is a way to watch Youtube videos and have a subscription feed using invidious without having a Google account. Google tracks you around, invidious does not and invidious is open source!

The best way to implement such a feature is within the Dissenter extension and not for Invidious since your unable to add a comment within the Invidious Dissenter commenting Iframe without logging into Dissenter.com. Also if Invidious wanted to remove Dissenter YouTube comment from their site this feature will circumvent that.

Another suggestion which would should not be hard to do would be to create a special URL for the dissenter extension that when you click on it it will open the URL within the dissenter extensions. For example...

dissenter://begin?url=https://www.youtube.com/watch?v=QIqZNh78E-A

And when you click on this URL it will open this page within the dissenter extension and allow you to comment on that URL. And all invidious and other sites needs to do is add the dissenter:// URL onto their website.

https://dissenter.com/discussion/begin?url=https://www.youtube.com/watch?v=QIqZNh78E-A

**Comment by justagoodperson** [comment]
The problem I see is that this is such an edge case (where you want to comment on a different URL than the one loaded because you're using a wrapper around a service) that the devs will never get around to implementing it. My suggestion would be to code it yourself for your own uses and submit a PR, and even still it likely wouldn't be added because it's too small of an edge case to justify adding to the codebase and supporting, probably you will just have to run your own fork, that's my honest assessment. Your best bet in terms of time and simplicity is to copy the link to YouTube from Inviduous and paste it in the Dissenter.com web portal to access the comment section.

**Comment by trymeouteh** [comment]
That is true. I made this suggestion so that I will not ever have to connect to youtube.com since I can use Invidious and still get youtube.com URL comments in Dissenter. The reason I want to avoid youtube.com was because of Googles creepy tracking and ads. I know Dissenter has a browser that does that and use Brave which is Dissenter is a fork of, but you never know what Google can get passed the ad blockers when it comes to tracking.

**Comment by ghost** [comment]
If gab moves to mastodon, could we use mastodon comment system? Add federation to dissenter and add federation to invidious?
